File: Programmers/java/lev0/구슬을 나누는 경우의 수/solution.java
import java.math.BigInteger;
class Solution {
    public int solution(int ball, int share) {
        share = Math.max(share, ball-share);
        BigInteger answer= BigInteger.valueOf(1);

          for (int i = ball;i>share ; i--)
              answer = answer.multiply(BigInteger.valueOf(i));
          for (int i = 1;i<=ball-share;i++)
              answer = answer.divide(BigInteger.valueOf(i));

        return answer.intValueExact();
    }
}

File: Programmers/py/lev0/구슬을 나누는 경우의 수/solution.py
# 간단 버전
import math
def solution(balls, share):
    return math.comb(balls,share)

# 구현
def solution(balls, share):
    parent,child = 1,1
    share = max(share, balls-share)
    
    for i in range(share+1,balls+1):
        parent*=i
    for i in range(1,balls-share+1):
        child *= i 
        
    return int(parent/child)